### PR TITLE
Param reassigned

### LINKED ISSUE #489: Method parameters should not be reassigned
RSPEC: https://jira.sonarsource.com/browse/RSPEC-1226

### PR DESCRIPTION
**Pull request by tvsonar**
Fixes #489

### CODE REVIEW

**Comment by dbolkensteyn** [comment]
@tvsonar My experimental VS crashed as soon as I opened it

File: src/SonarLint/Rules/ParameterAssignedTo.cs [hunk 1]
```diff
@@ -84,34 +50,105 @@ public override void Initialize(AnalysisContext context)
...
+                // https://github.com/dotnet/roslyn/issues/6209
+                // so:
+                var location = localSymbol.Locations.FirstOrDefault();
+                if (location == null)
+                {
+                    return false;
+                }
+
+                var declarationName = node.SyntaxTree.GetRoot().FindNode(location.SourceSpan, getInnermostNodeForTie: true) as IdentifierNameSyntax;
+                var catchStatement = declarationName.Parent as CatchStatementSyntax;
```
Comment (dbolkensteyn) [inline src/SonarLint/Rules/ParameterAssignedTo.cs]: @tvsonar I'm getting a NullReferenceException here

Comment (tvsonar) [inline src/SonarLint/Rules/ParameterAssignedTo.cs]: @dbolkensteyn I can't reproduce this, but added the null check.

**Comment by dbolkensteyn** [comment]
LGTM

**Pull request merged**